src/modules/lv2/factory.c

```diff
@@ -0,0 +1,261 @@
+                                        NULL);
+                snprintf(key, sizeof(key), "%d[*]", j);
+                mlt_properties_set(p, "identifier", key);
+                add_port_to_metadata(p, desc, j);
+                mlt_properties_set(p, "readonly", "yes");
+            }
+
+            p = mlt_properties_new();
+            snprintf(key, sizeof(key), "%d", mlt_properties_count(params));
+            mlt_properties_set_data(params, key, p, 0, (mlt_destructor) mlt_properties_close, NULL);
+            mlt_properties_set(p, "identifier", "instances");
+            mlt_properties_set(p, "title", "Instances");
+            mlt_properties_set(p,
+                               "description",
+                               "The number of instances of the plugin that are in use.\n"
+                               "MLT will create the number of plugins that are required "
+                               "to support the number of audio channels.\n"
+                               "Status parameters (readonly) are provided for each instance "
+                               "and are accessed by specifying the instance number after the "
+                               "identifier (starting at zero).\n"
+                               "e.g. 9[0] provides the value of status 9 for the first instance.");
+            mlt_properties_set(p, "type", "integer");
+            mlt_properties_set(p, "readonly", "yes");
+
+            if (type == mlt_service_filter_type) {
+
+                p = mlt_properties_new();
+                snprintf(key, sizeof(key), "%d", mlt_properties_count(params));
+                mlt_properties_set_data(params,
+                                        key,
+                                        p,
+                                        0,
+                                        (mlt_destructor) mlt_properties_close,
+                                        NULL);
+                mlt_properties_set(p, "identifier", "wetness");
+                mlt_properties_set(p, "title", "Wet/Dry");
+                mlt_properties_set(p, "type", "float");
+                mlt_properties_set_double(p, "default", 1);
+                mlt_properties_set_double(p, "minimum", 0);
+                mlt_properties_set_double(p, "maximum", 1);
+                mlt_properties_set(p, "mutable", "yes");
+                mlt_properties_set(p, "animation", "yes");
+            }
+        }
+    }
+#endif
+
+    return result;
+}
+
+MLT_REPOSITORY
+{
+
+#ifdef GPL
+    GSList *list;
+
+    g_lv2_plugin_mgr = lv2_mgr_new();
+
+    for (list = g_lv2_plugin_mgr->all_plugins; list; list = g_slist_next(list)) {
+
+      lv2_plugin_desc_t *desc = (lv2_plugin_desc_t *) list->data;
+      char *s = NULL;
+      s = calloc(1, strlen("lv2.") + strlen (desc->uri) + 1);
+
+      sprintf(s, "lv2.%s", desc->uri);
+
+      char *str_ptr = strchr(s, ':');
+      while (str_ptr != NULL)
+	{
+	  *str_ptr++ = '<';
+	  str_ptr = strchr(str_ptr, ':');
+	}
```

Comment:
Regarding the plugin name, since it must be altered, please drop the broken scheme and just make it like "lv2//kxstudio.sf.net/carla/plugins/audiogain_s". Then, we can just say it is the schema-less URI. Notice I did not include a period after "lv2" because the "//" is enough delimiters but still following some URI grammar. Or, since we are breaking URI grammar use  "lv2.kxstudio.sf.net/carla/plugins/audiogain_s". Your choice between the two, but I would rather not hear from people trying to use "http:" either in support issues or as bug reports.

Reply:
As we said `:` can't be used in filter name for some reason.

Not all lv2 URIs contain http:// for example this one don't:
urn:distrho:vitalium

and `<` is disallowed in URI sytax so it can't conflict I there is chance there is another plugin URI with similar name but different delimiter and mlt/melt accept it so I choose it.

but still this is ugly there should be another solution.

Reply:
The problem with `<` is that it is not friendly on the command line, and it is reserved character in XML. These are the 2 main protocols for MLT! How about `^`?

Reply:
yeah I got the point it would be good for xml and commandline and also (^) is in url forbidden characters so it should be fine.

https://url.spec.whatwg.org/#host-miscellaneous
http://www.faqs.org/rfcs/rfc1738.html